[C] PostgreSQL driver does not build on macOS 10.9

```
In file included from /Users/runner/mambaforge/conda-bld/arrow-adbc-split_1672320127369/work/c/driver/postgresql/connection.cc:26:
/Users/runner/mambaforge/conda-bld/arrow-adbc-split_1672320127369/work/c/driver/postgresql/util.h:130:10: error: use of undeclared identifier 'ntohll'; did you mean 'atoll'?
  return ntohll(v);
         ^~~~~~
         atoll
```

From what I see, this only started being available in 10.10

--- a/c/driver/postgresql/util.h
+++ b/c/driver/postgresql/util.h
@@ -40,12 +40,15 @@ namespace adbcpq {
 #define CONCAT(x, y) x##y
 #define MAKE_NAME(x, y) CONCAT(x, y)
 
-#if defined(__linux__)
-static inline uint64_t SwapNetworkToHost(uint64_t x) { return be64toh(x); }
-static inline uint64_t SwapHostToNetwork(uint64_t x) { return htobe64(x); }
+#if defined(_WIN32)
+static inline uint64_t SwapNetworkToHost(uint64_t x) { return ntohll(x); }
+static inline uint64_t SwapHostToNetwork(uint64_t x) { return htonll(x); }
 #elif defined(__APPLE__)
 static inline uint64_t SwapNetworkToHost(uint64_t x) { return OSSwapBigToHostInt64(x); }
 static inline uint64_t SwapHostToNetwork(uint64_t x) { return OSSwapHostToBigInt64(x); }
+#else
+static inline uint64_t SwapNetworkToHost(uint64_t x) { return be64toh(x); }
+static inline uint64_t SwapHostToNetwork(uint64_t x) { return htobe64(x); }
 #endif
 
 // see arrow/util/string_builder.h

--- a/c/driver/postgresql/util.h
+++ b/c/driver/postgresql/util.h
@@ -41,11 +41,11 @@ namespace adbcpq {
 #define MAKE_NAME(x, y) CONCAT(x, y)
 
 #if defined(__linux__)
-static inline uint64_t ntohll(uint64_t x) { return be64toh(x); }
-static inline uint64_t htonll(uint64_t x) { return htobe64(x); }
-#elif defined(__APPLE__) && !defined(ntohll)
-static inline uint64_t ntohll(uint64_t x) { return OSSwapBigToHostInt64(x); }
-static inline uint64_t htonll(uint64_t x) { return OSSwapHostToBigInt64(x); }
+static inline uint64_t SwapNetworkToHost(uint64_t x) { return be64toh(x); }
+static inline uint64_t SwapHostToNetwork(uint64_t x) { return htobe64(x); }
+#elif defined(__APPLE__)
+static inline uint64_t SwapNetworkToHost(uint64_t x) { return OSSwapBigToHostInt64(x); }
+static inline uint64_t SwapHostToNetwork(uint64_t x) { return OSSwapHostToBigInt64(x); }
 #endif
 
 // see arrow/util/string_builder.h
@@ -132,7 +132,7 @@ static inline uint32_t LoadNetworkUInt32(const char* buf) {
 static inline int64_t LoadNetworkUInt64(const char* buf) {
   uint64_t v = 0;
   std::memcpy(&v, buf, sizeof(uint64_t));
-  return ntohll(v);
+  return SwapNetworkToHost(v);
 }
 
 static inline int32_t LoadNetworkInt32(const char* buf) {
@@ -144,7 +144,7 @@ static inline int64_t LoadNetworkInt64(const char* buf) {
 }
 
 static inline uint64_t ToNetworkInt64(int64_t v) {
-  return htonll(static_cast<uint64_t>(v));
+  return SwapHostToNetwork(static_cast<uint64_t>(v));
 }
 
 }  // namespace adbcpq

--- a/c/driver/postgresql/util.h
+++ b/c/driver/postgresql/util.h
@@ -29,6 +29,8 @@
 
 #if defined(__linux__)
 #include <endian.h>
+#elif defined(__APPLE__)
+#include <libkern/OSByteOrder.h>
 #endif
 
 #include "adbc.h"
@@ -41,6 +43,9 @@ namespace adbcpq {
 #if defined(__linux__)
 static inline uint64_t ntohll(uint64_t x) { return be64toh(x); }
 static inline uint64_t htonll(uint64_t x) { return htobe64(x); }
+#elif defined(__APPLE__) && !defined(ntohll)
+static inline uint64_t ntohll(uint64_t x) { return OSSwapBigToHostInt64(x); }
+static inline uint64_t htonll(uint64_t x) { return OSSwapHostToBigInt64(x); }
 #endif
 
 // see arrow/util/string_builder.h

--- a/ci/scripts/glib_test.sh
+++ b/ci/scripts/glib_test.sh
@@ -34,15 +34,12 @@ test_subproject() {
         export GI_TYPELIB_PATH="${GI_TYPELIB_PATH}:${CONDA_PREFIX}/lib/girepository-1.0"
         export PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:${CONDA_PREFIX}/lib/pkgconfig"
     fi
+    # Required to avoid errors relating to symbols not available on
+    # older macOS (which the build will target by default)
+    export MACOSX_DEPLOYMENT_TARGET=10.14
 
     pushd "${source_dir}/glib"
     echo "Testing GLib"
-
-    if [[ "$(uname)" = "Darwin" ]]; then
-        bundle config build.red-arrow -- \
-               --with-cflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\" \
-               --with-cppflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\"
-    fi
 
     bundle install
     bundle exec \
@@ -54,12 +51,6 @@ test_subproject() {
     pushd "${source_dir}/ruby"
     echo "Testing Ruby"
 
-    if [[ "$(uname)" = "Darwin" ]]; then
-        bundle config build.red-arrow -- \
-               --with-cflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\" \
-               --with-cppflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\"
-    fi
-
     bundle install
     bundle exec \
            env DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \
@@ -67,12 +58,7 @@ test_subproject() {
     bundle exec rake build
 
     echo "Testing Gem"
-    local gem_flags=""
-    if [[ "$(uname)" = "Darwin" ]]; then
-        gem_flags='-- --with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
-    fi
-
-    gem install --install-dir "${build_dir}/gems" pkg/*.gem -- ${gem_flags}
+    gem install --install-dir "${build_dir}/gems" pkg/*.gem
     popd
 }
 

--- a/ci/scripts/glib_test.sh
+++ b/ci/scripts/glib_test.sh
@@ -72,7 +72,7 @@ test_subproject() {
         gem_flags='-- --with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
     fi
 
-    gem install --install-dir "${build_dir}/gems" pkg/*.gem ${gem_flags}
+    gem install --install-dir "${build_dir}/gems" pkg/*.gem -- ${gem_flags}
     popd
 }
 

--- a/ci/scripts/glib_test.sh
+++ b/ci/scripts/glib_test.sh
@@ -36,6 +36,7 @@ test_subproject() {
     fi
 
     pushd "${source_dir}/glib"
+    echo "Testing GLib"
 
     if [[ "$(uname)" = "Darwin" ]]; then
         bundle config build.red-arrow -- \
@@ -51,6 +52,7 @@ test_subproject() {
 
     export GI_TYPELIB_PATH="${install_dir}/lib/girepository-1.0:${GI_TYPELIB_PATH}"
     pushd "${source_dir}/ruby"
+    echo "Testing Ruby"
 
     if [[ "$(uname)" = "Darwin" ]]; then
         bundle config build.red-arrow -- \
@@ -63,7 +65,14 @@ test_subproject() {
            env DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \
            ruby test/run.rb
     bundle exec rake build
-    gem install --install-dir "${build_dir}/gems" pkg/*.gem
+
+    echo "Testing Gem"
+    local gem_flags=""
+    if [[ "$(uname)" = "Darwin" ]]; then
+        gem_flags='--with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
+    fi
+
+    gem install --install-dir "${build_dir}/gems" pkg/*.gem ${gem_flags}
     popd
 }
 

--- a/ci/scripts/glib_test.sh
+++ b/ci/scripts/glib_test.sh
@@ -34,12 +34,15 @@ test_subproject() {
         export GI_TYPELIB_PATH="${GI_TYPELIB_PATH}:${CONDA_PREFIX}/lib/girepository-1.0"
         export PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:${CONDA_PREFIX}/lib/pkgconfig"
     fi
-    # Required to avoid errors relating to symbols not available on
-    # older macOS (which the build will target by default)
-    export MACOSX_DEPLOYMENT_TARGET=10.14
 
     pushd "${source_dir}/glib"
     echo "Testing GLib"
+
+    if [[ "$(uname)" = "Darwin" ]]; then
+        bundle config build.red-arrow -- \
+               --with-cflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\" \
+               --with-cppflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\"
+    fi
 
     bundle install
     bundle exec \
@@ -51,6 +54,12 @@ test_subproject() {
     pushd "${source_dir}/ruby"
     echo "Testing Ruby"
 
+    if [[ "$(uname)" = "Darwin" ]]; then
+        bundle config build.red-arrow -- \
+               --with-cflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\" \
+               --with-cppflags=\"-D_LIBCPP_DISABLE_AVAILABILITY\"
+    fi
+
     bundle install
     bundle exec \
            env DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \
@@ -58,7 +67,12 @@ test_subproject() {
     bundle exec rake build
 
     echo "Testing Gem"
-    gem install --install-dir "${build_dir}/gems" pkg/*.gem
+    local gem_flags=""
+    if [[ "$(uname)" = "Darwin" ]]; then
+        gem_flags='-- --with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
+    fi
+
+    gem install --install-dir "${build_dir}/gems" pkg/*.gem -- ${gem_flags}
     popd
 }
 

--- a/ci/scripts/glib_test.sh
+++ b/ci/scripts/glib_test.sh
@@ -69,7 +69,7 @@ test_subproject() {
     echo "Testing Gem"
     local gem_flags=""
     if [[ "$(uname)" = "Darwin" ]]; then
-        gem_flags='--with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
+        gem_flags='-- --with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
     fi
 
     gem install --install-dir "${build_dir}/gems" pkg/*.gem ${gem_flags}

